src/networking.c

```diff
@@ -379,7 +381,8 @@ void putClientInPendingWriteQueue(client *c) {
     if (!c->flag.pending_write &&
         (!c->repl_data ||
          c->repl_data->repl_state == REPL_STATE_NONE ||
-         (isReplicaReadyForReplData(c) && !c->repl_data->repl_start_cmd_stream_on_ack))) {
+         (isReplicaReadyForReplData(c) && !c->repl_data->repl_start_cmd_stream_on_ack)) &&
+        (!c->slot_migration_job || clusterSlotMigrationShouldInstallWriteHandler(c))) {
```

Comment:
checking `c->slot_migration_job` in this function leaks the implementation details of `clusterSlotMigrationShouldInstallWriteHandler`. maybe either move it into `clusterSlotMigrationShouldInstallWriteHandler` or perhaps formalize the notion of a slot migration client by wrapping it under `isClusterSlotExportClient`? my preference is the former though. along the same line, I think the existing check of `!c->repl_data ||
         c->repl_data->repl_state == REPL_STATE_NONE` is a leaky abstraction as well. there should've be just one function like `replicaShouldInstallWriterHandler(c)` imo that performs all the magic checks.

```suggestion
        clusterSlotMigrationShouldInstallWriteHandler(c)) {
```

Reply:
Agree. Yeah it is hard when the code you are reading/matching style to is not a good guide. Updated to move it into clusterSlotMigrationShouldInstallWriteHandler